backend/error_source.go

```diff
@@ -21,10 +21,20 @@ const (
 	DefaultErrorSource ErrorSource = ErrorSourcePlugin
 )
 
+// IsValid return true if es is ErrorSourceDownstream or ErrorSourcePlugin.
 func (es ErrorSource) IsValid() bool {
```

Comment:
Are there cases `es` is something different from both, and we discard the error or do something else based on the source? With the `IsValid` check, we are making sure that we need to handle, right? Can you please provide a scenario? Asking for learning purpose. :lab_coat:

Reply:
In theory it could be since you could do `ErrorSource("")`. This could happen here when converting from protobuf (response sent from a backend plugin to grafana) without populating the `ErrorSource` field. https://github.com/grafana/grafana-plugin-sdk-go/blob/6e354283d25643562860e743fb53ddd9399c2e2d/backend/convert_from_protobuf.go#L162

Reply:
Or https://github.com/grafana/grafana-plugin-sdk-go/blob/6e354283d25643562860e743fb53ddd9399c2e2d/backend/data_adapter.go#L48-L51 is maybe a better example where the plugin author haven't set the `ErrorSource` and we want to enforce setting it.